components/constellation/constellation.rs

```diff
@@ -1973,6 +1993,10 @@ impl<Message, LTF, STF> Constellation<Message, LTF, STF>
             self.update_browsing_context(browsing_context_id, pipeline_id);
         }
 
+        for (pipeline_id, history_state_id) in pipeline_changes.drain() {
+            self.update_pipeline(pipeline_id, history_state_id);
+        }
+
```

Comment:
Should we do this before updating the browsing contexts? Otherwise we might get a glitch where the browsing context's current pipeline has been updated but not its state.

Reply:
I'm not sure how to avoid this, but I would rather the pipeline's activities are all properly set before updating the session history state so that the `popstate` event will run when the document is active.